src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1304,6 +1304,7 @@ LedgerManagerImpl::applyTransactions(
                    tx->getSeqNum(),
                    mApp.getConfig().toShortString(tx->getSourceID()));
         tx->apply(mApp, ltx, tm);
+        tx->processPostApply(mApp, ltx, tm);
```

Comment:
I think  it's better to perform this `PostApply` there so that changes get merged as needed: https://github.com/stellar/stellar-core/blob/d9ecb7e727eb24a046bafc7878ef22466c268e71/src/transactions/TransactionFrame.cpp#L1349

if you do it like this, you end up with the same ledger entry (the account paying for the fee) being modified twice in the same `LedgerEntryChanges` if the code at the other location does something (signers are in the same ledger entry), which wastes resources and may potentially confuse downstream systems.

Reply:
Currently the only other case for changes after is pre-V10, so there is no conflict with Soroban. Going forward I would actually prefer to have post-apply changes to be called post-apply and not as a part of it. E.g. in this case if we put the refund logic inside apply, then we might omit refund in case of internal tx error (which is of course unlikely, but semantically it still doesn't seem right to not do the refund in such case). 

FWIW I've also tried to put this into apply, but besides the awkward failure cases, it is also inconvenient for tests - we charge fees and apply txs separately and hence if the test just applies the tx without charging the fee, it might produce some XLM, which is confusing.

Reply:
yes makes sense thank you